src/v/s3/client.cc

```diff
@@ -527,20 +531,31 @@ ss::future<> client_pool::stop() {
 ///         are in use)
 ss::future<client_pool::client_lease> client_pool::acquire() {
     gate_guard guard(_gate);
-    if (_pool.empty()) {
-        if (_policy == client_pool_overdraft_policy::wait_if_empty) {
-            co_await _cvar.wait([this] { return !_pool.empty(); });
-        } else {
-            auto cl = ss::make_shared<client>(_config, _as);
-            _pool.emplace_back(std::move(cl));
+    try {
+        while (_pool.empty() && !_gate.is_closed()) {
+            if (_policy == client_pool_overdraft_policy::wait_if_empty) {
+                co_await _cvar.wait();
+            } else {
+                auto cl = ss::make_shared<client>(_config, _as);
+                _pool.emplace_back(std::move(cl));
+            }
         }
+    } catch (const ss::broken_condition_variable&) {
     }
+    if (_gate.is_closed() || _as.abort_requested()) {
+        throw ss::gate_closed_exception();
```

Comment:
nit: just return the exceptional future? no need to walk through the global lock to convert to exceptional future. 

ok to postpone as well.